File: MAP/C# - Toy language/Lab10/Toy Language/Toy Language/model/expressions/VarExp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language.utils;

namespace Toy_Language.model.expressions
{
    public class VarExp : Exp
    {
        private string id;

        public VarExp(string _id)
        {
            this.id = _id;
        }

        public override int Eval(IModDictionary<string, int> symbTable)
        {
            if (symbTable.Contains(id))
            {
                return symbTable.Get(id);
            }

            throw new Exception("Element not found");
        }

        public override string ToString()
        {
            return id;
        }
    }
}

File: Year 1/MAP/JavaRepo - Toy language/lab2/src/model/Expressions/VarExp.java
package model.Expressions;

import exceptions.NotExistingException;
import utils.adt.MyIDictionary;
import utils.adt.MyIHeap;

public class VarExp extends Exp {
    String id;

    public VarExp(String _id){
        id = _id;
    }

    @Override
    public int eval(MyIDictionary<String, Integer> tbl,
                    MyIHeap<Integer, Integer> heap) throws NotExistingException
    {
        if (tbl.contains(id)){
            return tbl.get(id);
        }
        throw new NotExistingException("Not found");
    }

    @Override
    public String toString() {
        return id;
    }
}
